### PR TITLE
nits

### PR DESCRIPTION
**Pull request by dhrubabasu**
* Removes `txBuilder` from `VM`
* Removes `atomicUtxosManager` from `VM`

### CODE REVIEW

File: vms/platformvm/service.go [hunk 1]
```diff
@@ -381,7 +381,8 @@ func (s *Service) GetUTXOs(_ *http.Request, args *api.GetUTXOsArgs, response *ap
 			limit,
 		)
 	} else {
-		utxos, endAddr, endUTXOID, err = s.vm.atomicUtxosManager.GetAtomicUTXOs(
```
Comment (abi87) [inline vms/platformvm/service.go:384]: can we create this only once when we instantiace the `service` struct?

File: vms/platformvm/vm.go [hunk 2]
```diff
@@ -85,8 +83,7 @@ type VM struct {
 	// Bootstrapped remembers if this chain has finished bootstrapping or not
 	bootstrapped utils.Atomic[bool]
 
-	txBuilder txbuilder.Builder
```
Comment (abi87) [inline vms/platformvm/vm.go:88]: no client is using txBuilder, so I agree we should get rid of it as a vm member

File: vms/platformvm/vm.go [hunk 3]
```diff
@@ -65,8 +64,7 @@ type VM struct {
 	*network.Network
 	validators.State
 
-	metrics            metrics.Metrics
-	atomicUtxosManager avax.AtomicUTXOManager
```
Comment (abi87) [inline vms/platformvm/vm.go:69]: only `service` uses `atomicUtxosManager`, so `atomicUtxosManager` should be a service member

File: vms/platformvm/vm_regression_test.go [hunk 4]
```diff
@@ -47,11 +47,12 @@ import (
 	"github.com/ava-labs/avalanchego/vms/secp256k1fx"
 
 	blockexecutor "github.com/ava-labs/avalanchego/vms/platformvm/block/executor"
+	txbuilder "github.com/ava-labs/avalanchego/vms/platformvm/txs/builder"
 )
 
 func TestAddDelegatorTxOverDelegatedRegression(t *testing.T) {
 	require := require.New(t)
-	vm, _, _ := defaultVM(t, cortina)
+	vm, txBuilder, _, _ := defaultVM(t, cortina)
```
Comment (abi87) [inline vms/platformvm/vm_regression_test.go:55]: this is the one bit slightly annoying to me: why `defaultVM` should return the `txBuilder`?
Can we instead create the `txBuilder` only in the tests where we need it and leave defaultVM signature unchanged?

**Review by abi87 — CHANGES_REQUESTED** [review]
I think it's a good change. I have only minor observations around handling txBuilder creation in tests.
Looking forward to merge this cleanup once we solve the comments!

File: vms/platformvm/vm_regression_test.go (hunk 4 above)
Comment (dhrubabasu) [reply vms/platformvm/vm_regression_test.go:55]: over 50% of the tests use txBuilder (including `defaultVM` itself). I think we should keep it in there

**Review by abi87 — APPROVED** [review]

**Pull request merged**